### PR TITLE
doc: use extlink for github issue links

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
Use extlink for GitHub issue links in documentation.

### CODE REVIEW

**Comment by henrikbrixandersen** [comment]
This is a hotfix for the documentation build issue seen in https://github.com/zephyrproject-rtos/zephyr/runs/4833163774 (#41847).

**Review by Thalley — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Comment by henrikbrixandersen** [comment]
This issue is a result of https://github.com/sphinx-doc/sphinx/pull/9800 which was merged in Sphinx v4.4.0.

**Review by erwango — APPROVED** [review]

File: doc/guides/west/why.rst [hunk 1]
```diff
@@ -117,7 +117,4 @@ West is:
   interoperability with third party tools, and means Zephyr developers
   can always find out what is happening "under the hood" when using west.
 
-See `Zephyr issue #6205`_ and for more details and discussion.
-
-.. _Zephyr issue #6205:
-   https://github.com/zephyrproject-rtos/zephyr/issues/6205
+See :github:`Zephyr issue #6205 <6205>`_ and for more details and discussion.
```
Comment (gmarull) [inline doc/guides/west/why.rst:120]:
```suggestion
See :github:`Zephyr issue #6205 <6205>` and for more details and discussion.
```

my mistake, sorry

Comment (henrikbrixandersen) [reply doc/guides/west/why.rst]: 👍

**Pull request merged**